reg/drone/service/battery/Parameters.0.1.uavcan

```diff
@@ -0,0 +1,54 @@
+# Smart battery parameter message. It is mostly intended for automated battery charging and maintenance systems.
+# This message is modeled after the Smart Battery Data Specification (SBS) and the MAVLink battery status messages.
+#
+# The values carried by this message are either constant or slow-changing, so, generally, the publishing frequency
+# should not be higher than 0.2 Hz, and the priority should be either OPTIONAL or SLOW.
+#
+# All parameters are required unless specifically stated otherwise.
+
+truncated uint64 unique_id
+# A statistically unique number that can be used to differentiate this exact battery from others.
+# Needed for logging purposes. If the battery supports SBS, this value may be computed as CRC64WE of
+# (ManufacturerName + DeviceName + ManufactureDate + SerialNumber).
+# This value shall be invariant to the identity of the reporting node unless it is an integral part of the battery.
+# Vendors may use a constant random number for the 40 most significant bits for vendor identification purposes.
+
+uavcan.si.unit.mass.Scalar.1.0 mass
+# The total mass of the battery, including the packaging, electronics, cabling, and all auxiliary items, if any.
+# May be used for predicting the kinematic parameters of the vehicle.
+# NaN if unknown.
+
+uavcan.si.unit.electric_charge.Scalar.1.0 design_capacity
+# The maximum total charge of the pack, at 100% SoH, specified by the manufacturer.
+
+uavcan.si.unit.voltage.Scalar.1.0[2] design_cell_voltage_min_max
+# The minimum (end of discharge) and the maximum (end of charge) cell voltages specified by the manufacturer
+# at 100% SoH. Example: {2.8, 4.2} V
+
+uavcan.si.unit.electric_current.Scalar.1.0 discharge_current
+uavcan.si.unit.electric_current.Scalar.1.0 discharge_current_burst      # At least for 5 seconds.
+uavcan.si.unit.electric_current.Scalar.1.0 charge_current
```

Comment:
Would I indicate a primary battery by setting charge_current, charge_current_fast, charge_termination_treshold to 0, and possibly selecting a battery technology of a primary chemistry? Thinking of lithium thionyl chloride batteries, carbon-zinc, metallic lithium batteries.

Reply:
I am not sure I understand how do you define "primary battery" here? A vehicle that is equipped with multiple batteries would have multiple Parameter subjects, one per battery. A subscriber that needs to know the parameters of a specific battery would subscribe to the correct subject and get the data from there.

One pertinent question is how do we auto-configure the subscriber such that the integrator doesn't have to, but this is not something that we will be working on in this first version of DS-015. This will be addressed in an incremental fashion later.

Reply:
By primary battery I mean non-rechargeable, as in https://en.wikipedia.org/wiki/Primary_cell

Reply:
Understood.. I just added a comment that for non-rechargeable batteries the parameters should be set to NaN.

Can you please provide a full list of new battery chemistry types that should be added to the Technology enumeration?

Reply:
Non-Rechargeable, used in maritime robotics and harsh environment for main power
* LI_SOCL2 - lithium / thionyl chloride (Li-SOCL2)
* LI_BCX - lithium / thionyl chloride + bromine chloride (Li-BCX)
* LI_MNO2 - Lithium / manganese dioxide (Li-MnO2) (eg lithium coin cell, lithium 9V)
* ZN_O2  - Zinc-Air
* AL_O2 - Aluminum-Air

Non-Rechargeable, commodity. a robot might not use these for main power, but a subsystem might (payload, comms)
* ZN_MNO2_NH4CL - zinc / manganese dioxide - ammonium chloride electrolyte (aka zinc-carbon)
* ZN_MNO2_ZNCL2 - zinc / manganese dioxide - Zinc chloride electrolyte     (aka heavy duty zinc-carbon)
* ZN_MNO2_KOH   - zinc / manganese dioxide - Potassium hydroxide electrolyte (aka alkaline)

Rechargable:
* LI_S - lithium / sulfur (LiS) - starting to be available

Supercaps
* EDLC - Electrostatic double-layer capacitors, large arrays benefit from charge management similar to batteries with balancing and current limited charge.

Reply:
Thanks, done.